### DISCUSSION TITLE
Unable to create custom-fields via API

### DISCUSSION THREAD

**Discussion by alop** [netbox-community/netbox#5713] [Q&A] [answered]
### Environment
* Python version: 3.8
* NetBox version: 2.10.3

### Steps to Reproduce
1. Attempt to create a custom-field
2. 
```
curl -H "content-type: application/json" -H "accept: application/json" -H "Authorization: <TOKEN> -X POST "https://netbox/api/extras/custom-fields/" -d '{"content_types":["ipam.ipaddress"],"name":"test_session_id","type":{"value":"text","label":"Text"},"label":"Test Session Index","description":"Index of the session to which IP address is assigned.","required":true,"default":null}'
```
3. 

### Expected Behavior
Expected a 201 response about a new custom-field being created

### Observed Behavior
400 Bad Request
`{"type":["Value must be passed directly (e.g. \"foo\": 123); do not use a dictionary or list."]}`

**Comment by jeremystretch** [comment, marked as answer]
The error message is telling you exactly what's wrong:

> Value must be passed directly (e.g. "foo": 123); do not use a dictionary or list.

Try `"type": "text"`.

**Reply by alop** [reply]
Thanks, was confused by the structure of the `GET` results, which has `type: { value: "text", label: "text"}`
